lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -35,6 +36,30 @@ public enum S3SseType
         NONE, S3, KMS
     }
 
+    public enum ObjectCannedAcl
+    {
+        NONE(null),
+        PRIVATE(ObjectCannedACL.PRIVATE),
+        PUBLIC_READ(ObjectCannedACL.PUBLIC_READ),
+        PUBLIC_READ_WRITE(ObjectCannedACL.PUBLIC_READ_WRITE),
+        AUTHENTICATED_READ(ObjectCannedACL.AUTHENTICATED_READ),
+        AWS_EXEC_READ(ObjectCannedACL.AWS_EXEC_READ),
+        BUCKET_OWNER_READ(ObjectCannedACL.BUCKET_OWNER_READ),
+        BUCKET_OWNER_FULL_CONTROL(ObjectCannedACL.BUCKET_OWNER_FULL_CONTROL);
+
+        private final ObjectCannedACL objectCannedACL;
```

Comment:
Rename this and the constructor parameter to `cannedAcl` to make Error Prone is happy